cmd/minishift/cmd/start.go

```diff
@@ -154,42 +128,50 @@ func runStart(cmd *cobra.Command, args []string) {
 	}
 
 	//Create the host directories if not present
-	hostDirs := []string{viper.GetString(startFlags.HostConfigDir.Name),
-		viper.GetString(startFlags.HostDataDir.Name), viper.GetString(startFlags.HostVolumeDir.Name),
-		viper.GetString(startFlags.HostPvDir.Name)}
-	err = clusterup.EnsureHostDirectoriesExist(libMachineClient, hostDirs)
-	if err != nil {
-		fmt.Println("Error creating required host directories: ", err)
-		atexit.Exit(1)
+	hostDirs := []string{
+		viper.GetString(startFlags.HostConfigDir.Name),
+		viper.GetString(startFlags.HostDataDir.Name),
+		viper.GetString(startFlags.HostVolumeDir.Name),
+		viper.GetString(startFlags.HostPvDir.Name),
 	}
-
-	ip, err := host.Driver.GetIP()
+	err = clusterup.EnsureHostDirectoriesExist(hostVm, hostDirs)
 	if err != nil {
-		fmt.Println("Error determining host ip: ", err)
-		atexit.Exit(1)
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating required host directories: %v", err))
 	}
 
-	if proxyConfig.IsEnabled() {
-		// once we know the IP, we need to make sure it is not proxied in a proxy environment
-		proxyConfig.AddNoProxy(ip)
-		proxyConfig.ApplyToEnvironment()
+	autoMountHostFolders(hostVm.Driver)
+
+	openShiftVersion := clusterup.DetermineOpenShiftVersion(viper.GetString(startFlags.OpenshiftVersion.Name))
+	clusterUpConfig := &clusterup.ClusterUpConfig{
+		OpenShiftVersion: openShiftVersion,
+		MachineName:      constants.MachineName,
+		Ip:               ip,
+		Port:             constants.APIServerPort,
+		RoutingSuffix:    viper.GetString(startFlags.RoutingSuffix.Name),
+		HostPvDir:        viper.GetString(startFlags.HostPvDir.Name),
+		User:             defaultUser,
+		Project:          defaultProject,
+		KubeConfigPath:   constants.KubeConfigPath,
+		OcPath:           cacheOc(openShiftVersion),
```

Comment:
I see yeha, maybe the `cacheOc` should not happen here. It is a function that does too much